Dockerfile

```
@@ -2,8 +2,10 @@ FROM registry.access.redhat.com/ubi8/go-toolset:1.16.12-4 as build
 USER root
 RUN dnf install -y --setopt=tsflags=nodocs git
 COPY . /go/src/
-RUN cd /go/src/ && env GOOS=linux GOARCH=amd64 make build
-
+RUN ARCH=$(uname -m) ; echo $ARCH \
+    ; if [ "$ARCH" == "x86_64" ] ; then ARCH="amd64" ; elif [ "$ARCH" == "s390x" ] ; then ARCH="s390x" ; fi \
+    ; cd /go/src/ && env GOOS=linux GOARCH=${ARCH} make build \
+    ; 
```

```suggestion
RUN cd /go/src/ && make build
```
Why do you need GOOS and GOARCH?

---

Thats how Go uses the current platform’s GOOS and GOARCH to determine how to build the binary. Incase of github workflow , we do pass the platform option . But if we wants to build dockerfile manually , we have to pass this option for go to build the binary

---

I will do the changes , as you have mentioned . But don't you think for manual dockerfile build , we do need to pass the option for go to build the binary